source/MaterialXRuntime/Private/Commands/PvtRenamePrimCmd.cpp

```diff
@@ -17,6 +20,25 @@ void PvtRenamePrimCmd::execute(RtCommandResult& result)
 {
     try
     {
+        // Validate that the prim exists.
+        RtPrim prim = _stage->getPrimAtPath(_path);
+        if (!prim)
+        {
+            result = RtCommandResult(false, "Given path '" + _path.asString() + " does not point to a prim in this stage");
+            return;
+        }
+
+        // Make sure the new name is a unique child name for the parent.
+        // Otherwise the name will be updated during rename in order to be
+        // unique, and the message we send will not have correct information.
+        RtPrim parent = prim.getParent();
+        PvtDataHandle parentH = PvtObject::hnd(parent);
+        PvtPrim* parentPtr = parentH->asA<PvtPrim>();
+        _newName = parentPtr->makeUniqueChildName(_newName);
```

Comment:
We must check that the new name used is a unique child name in the parent. Otherwise it will be renamed to be unique, and then the message we send below will not contain the correct name.